Sources/SwiftInspectorVisitors/Tests/AssociatedtypeVisitorSpec.swift

```diff
@@ -0,0 +1,136 @@
+// Created by Dan Federman on 2/17/21.
+//
+// Copyright © 2021 Dan Federman
+//
+// Distributed under the MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import Nimble
+import Quick
+import SwiftInspectorTestHelpers
+
+@testable import SwiftInspectorVisitors
+
+final class AssociatedtypeVisitorSpec: QuickSpec {
+  private var sut = AssociatedtypeVisitor()
+
+  override func spec() {
+    beforeEach {
+      self.sut = AssociatedtypeVisitor()
+    }
+
+    describe("visit(_:)") {
+      context("visiting a protocol with associated types") {
+        var associatedTypeNameToInfoMap: [String: AssociatedtypeInfo]?
+        beforeEach {
+          let content = """
+            public protocol Collection {
```

Comment:
I straight stole Swift's `Collection` declaration and associatedtypes since they were immensely complicated.